homework/index.js

```diff
@@ -1,46 +1,99 @@
 'use strict';
 
-{
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
+function fetchJSON(url, cb) {
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  xhr.responseType = 'json';
+  xhr.onload = () => {
+    if (xhr.status >= 200 && xhr.status <= 299) {
+      cb(null, xhr.response);
+    } else {
+      cb(new Error(`Network error:${xhr.status} -${xhr.statusText}`));
+    }
+  };
+  xhr.onerror = () => cb(new Error('Network request failed'));
+  xhr.send();
+}
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
+function createAndAppend(name, parent, options = {}) {
+  const elem = document.createElement(name);
+  parent.appendChild(elem);
+  Object.keys(options).forEach(key => {
+    const value = options[key];
+    if (key === 'text') {
+      elem.textContent = value;
+    } else {
+      elem.setAttribute(key, value);
+    }
+  });
+  return elem;
+}
+
+function main(url) {
+  fetchJSON(url, (err, repositories) => {
+    const root = document.getElementById('root');
+    if (err) {
+      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      return;
+    }
+    const repNames = [];
+    createAndAppend('select', root, { id: 'main-select' });
+    createAndAppend('div', root, { class: 'portion-parent', id: 'parent' });
+    const parent = document.getElementById('parent');
+    createAndAppend('div', parent, { class: 'left-portion', id: 'lft' });
+    createAndAppend('div', parent, { class: 'right-portion', id: 'rt' });
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
+    const leftPortion = document.getElementById('lft');
+    const rightPortion = document.getElementById('rt');
+    for (let i = 0; i < repositories.length; i++) {
+      const rep = repositories[i];
+      repNames.push(rep.name);
+    }
+    const selectItem = document.getElementById('main-select');
+
+    repNames.sort();
```

Comment:
Use `localeCompare` to sort the `repositories` array on the `name` property. This simple sort that you do here on just the names places all repositories that start with an uppercase letter above all repositories that start with a lowercase letter. People will expect the list to be sorted without taking case into account. `localeCompare` does this for you, as well as taking into accounted accented letter, as in French.